azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageTypeServlet.java

```diff
@@ -74,15 +87,74 @@ protected void handleGet(
       throws ServletException, IOException {
     try {
       final Map<String, String> templateVariableToValue = new HashMap<>();
-      if (IMAGE_TYPE_WITH_ID_URI_TEMPLATE.match(req.getRequestURI(), templateVariableToValue)) {
-        // TODO: Implementation will be provided in the future PR
+      if (IMAGE_TYPE_NAME_OR_ID_URI_TEMPLATE.match(req.getRequestURI(), templateVariableToValue)) {
+        getImageTypeDTOByIdOrImageTypeName(resp, session, templateVariableToValue);
       } else if (GET_IMAGE_TYPE_URI.equals(req.getRequestURI())) {
-        // TODO: Implementation will be provided in the future PR
+        ImageTypeDTO imageTypeDTO;
+        if (req.getQueryString().contains(ImageMgmtConstants.IMAGE_TYPE)) {
+          final String imageTypeName = HttpRequestUtils
+              .getParam(req, ImageMgmtConstants.IMAGE_TYPE);
+          if (!hasImageManagementPermission(imageTypeName, session.getUser(), Type.GET)) {
+            log.info(FORBIDDEN_USER_ERR_MSG);
+            throw new ImageMgmtInvalidPermissionException(ErrorCode.FORBIDDEN,
+                FORBIDDEN_USER_ERR_MSG);
+          }
+          imageTypeDTO = this.imageTypeService.findImageTypeWithOwnershipsByName(imageTypeName);
+          sendResponse(resp, HttpServletResponse.SC_OK, imageTypeDTO);
+        }
+        getAllImageTypeDTOs(resp, session);
+      } else {
+        log.info(PATH_NOT_SUPPORTED);
+        throw new ImageMgmtInvalidInputException(ErrorCode.NOT_FOUND, PATH_NOT_SUPPORTED);
       }
+    } catch (final ImageMgmtInvalidPermissionException e) {
+      log.error("The user provided does not have permissions to access the resource");
+      resp.setStatus(HttpStatus.SC_FORBIDDEN);
+      sendErrorResponse(resp, HttpServletResponse.SC_FORBIDDEN,
+          "User does not have permissions. Error Message: " + e.getMessage());
+    } catch (final ImageMgmtException e) {
+      if (e.getErrorCode() != null) {
+        log.error("An error has occurred");
+        resp.setStatus(e.getErrorCode().getCode());
+      } else {
+        resp.setStatus(HttpStatus.SC_BAD_REQUEST);
+      }
+      sendErrorResponse(resp, HttpServletResponse.SC_BAD_REQUEST,
+          "Exception on GET call to /imageTypes. Reason: " + e.getMessage());
     } catch (final Exception e) {
       log.error("Content is likely not present " + e);
       resp.setStatus(HttpStatus.SC_NOT_FOUND);
+      sendErrorResponse(resp, HttpServletResponse.SC_NOT_FOUND,
+          "Exception on GET call to /imageTypes. Reason: " + e.getMessage());
```

Comment:
Maybe we should revisit the Error Handling logic in a separate PR. In my opinion Internal Server Error (500) is more suitable than Not Found (404) when unexpected errors happen.

Reply:
Perhaps in future work, definitely!